src/linux.rs

```diff
@@ -21,6 +22,23 @@ pub fn linux_interfaces(
     Ok(Vec::from_iter(s))
 }
 
+pub fn posix_interfaces_by_index(
+    _display: InterfaceDisplay,
+) -> Result<IfacesByIndex, Box<dyn std::error::Error>> {
+    let mut interfaces = IfacesByIndex::new();
+
+    let iface_names_idxes = if_nameindex()?;
+
+    for iface in &iface_names_idxes {
+        interfaces.insert(
+            iface.index().try_into().unwrap(),
```

Comment:
Not convinced by the `unwrap()` here. This will cause a `panic!` (akin to python's `raise`) and I'm not sure how the layer between python and rust will handle it. If you pull it out of the insert, you can add `?` after the `try_into()` and you'll get the correct value, and the error will bubble up (again because of `Box<dyn ...>`)

```suggestion
let index = iface.index().try_into()?;
interfaces.insert(index, ...)
```

Reply:
Now I set it up to use `as` like the windows side does.  I'm still quite confused why you can't use `as` on a function argument (e.g. `interfaces.insert(index as uindex, ...)`).

Reply:
Yep, was able to build without issue. The conversion is required b/c of `c_uint` to `usize` but `c_uint <= size_t == usize` so it's always correct